### ISSUE TITLE
illegal hardware instruction error

### ISSUE THREAD

**Issue by heckj** [SwiftDocOrg/DocTest#8]
I installed swift-doctest with the homebrew formula, and added a sample markdown file to try this out. When I ran:

`swift-doctest SAMPLE.md`

I get the response:
`[1]    28640 illegal hardware instruction  swift-doctest SAMPLE.md`

which I'm guessing is a terribly crash in the swift-repl.

This is using Swift 5.2, the default built in with Catalina. `swift -version`:

```
Apple Swift version 5.2.2 (swiftlang-1103.0.32.6 clang-1103.0.32.51)
Target: x86_64-apple-darwin19.4.0
```

Any suggestions on debugging this or knowing where the issue might lie? I was hoping to use this on my repo's README.md to verify the snippets within it were functional... or at least that's what I got excited about, as I'm not entirely sure how to handle a package creation/import with those same statements.

**Comment by PaulSolt** [comment]
I also get illegal hardware instruction when I run the Swift file with Swift 5.2.

I installed manually using make.

**Add.swift**
```swift
/**
    Returns the sum of two integers.

    ```swift doctest
    add(1 1) // Returns 3.0
    ```
*/
func add(_ a: Int, _ b: Int) -> Int { 
	return a + b + 2
}

```

Terminal output with `zsh`

```bash
➜  DocTestExample swift doctest --package Add.swift
[1]    4729 illegal hardware instruction  swift doctest --package Add.swift
➜  DocTestExample swift doctest Add.swift          
[1]    4735 illegal hardware instruction  swift doctest Add.swift
```

Swift version 5.2
```bash
Apple Swift version 5.2 (swiftlang-1103.0.32.1 clang-1103.0.32.29)
Target: x86_64-apple-darwin19.4.0
```

**Comment by mattt** [comment]
@heckj @PaulSolt Thanks to both of you for reporting this. The fatal error is caused by `Can't form Range with upperBound < lowerBound`, which happens for examples that contain no assertions of expected input (@PaulSolt changing `// Returns 3.0` to `// => Double = 3.0` should get this to run correctly). I'd fixed an issue related to this in the underlying TAP library before, but perhaps I reintroduced that regression. Looking into this now.

**Edit**: Yep. Looks like my original patch in TAP got lost along the way. I was easily able to reproduce this in a unit test. Patching and re-releasing now.

**Comment by mattt** [comment]
Alright, this should now be fixed as of 0.0.2 (c02403705966011f81d3472eb2c1faa223cbd426). 

@heckj @PaulSolt Can you please confirm that this is now working for you?

**Comment by heckj** [comment]
@mattt yep - build from master, and "brew upgrade" result in happier end times. Thanks!

**Comment by mattt** [comment]
@heckj Glad to hear it! Thanks for confirming. Closing this now.

**Comment by PaulSolt** [comment]
@mattt Does this mean you can't use it to test a snippet of code to verify the code compiles without errors? (i.e. testing without an assertion)

I'd like to be able to verify code in a post or book is valid Swift code for the current version of Swift (or a specific version of Swift).

```swift
func viewDidLoad() {
    super.viewDidLoad()
    tableView.delegate = self        // defined and extended in a separate snippet
    tableView.dataSource = self
}
```

**Comment by mattt** [comment]
@PaulSolt To clarify, the crash you reported was due to a logic error when processing code without any assertions. With the fix in 0.0.2, it now runs successfully, reporting 0 tests run.

To answer your question: No, running doctest on code blocks without any assertions doesn't provide any real benefit right now. Syntax errors are communicated to the REPL through standard error, and won't be surfaced as failures except for things like `fatalError()` (and even then, I'm not 100% confident on the behavior; we still need tests for exceptional behavior like this). 

I'm inclined to think that the functionality you're describing is outside the scope of doctest, and would be better served by something else. Maybe that's a hypothetical `swift-syntaxcheck` tool, or maybe it's an approach where example projects are tested through conventional CI measures, and [transcluded](https://en.wikipedia.org/wiki/Transclusion) into markup by reference (e.g. `{% include code/example.swift#L1-10 %}`.
